packages/synapse-react-client/src/components/DirectDownload/DirectDownload.tsx

```diff
@@ -1,44 +1,38 @@
-import React, { useEffect, useState } from 'react'
+import React from 'react'
 import {
   BatchFileRequest,
-  FileEntity,
-  FileHandle,
+  ExternalFileHandle,
   FileHandleAssociateType,
   FileHandleAssociation,
 } from '@sage-bionetworks/synapse-types'
-import {
-  getFiles,
-  getFileResult,
-  getEntity,
-} from '../../synapse-client/SynapseClient'
+import { getFiles } from '../../synapse-client/SynapseClient'
 import IconSvg, { IconSvgProps } from '../IconSvg/IconSvg'
-import { useInView } from 'react-intersection-observer'
 import { useSynapseContext } from '../../utils/context/SynapseContext'
 import { TOOLTIP_DELAY_SHOW } from '../SynapseTable/SynapseTableConstants'
 import { UAParser } from 'ua-parser-js'
 import { Tooltip } from '@mui/material'
 import { implementsExternalFileHandleInterface } from '../../utils/types/IsType'
+import { useGetFileBatch } from '../../synapse-queries/file/useFiles'
 
 export type DirectFileDownloadProps = {
   associatedObjectId: string
-  entityVersionNumber?: string
```

Comment:
For this component I wanted to use react-query so I could pre-fill the cache in the SynapseTable case. However, we had complicated branching logic because you could either pass a full file handle association, or just pass an entity ID and version number and the component would magically figure out the file handle association for you.

IMO supporting both configurations in one component was confusing, especially when trying to integrate with react-query, so this was simplified to only accept a file handle association.

I don't think the FileEntity-specific case was adding much value, but if we think it's useful, we could add a custom component that uses this one, rather than intertwining logic for identifying the file handle association and using it.